transport/tchannel/channel_outbound.go

```diff
@@ -174,6 +174,10 @@ func (o *ChannelOutbound) Call(ctx context.Context, req *transport.Request) (*tr
 
 	res := call.Response()
 	headers, err := readHeaders(format, res.Arg2Reader)
+	// service name match validation, return yarpcerrors.CodeInternal error if not match
+	if match, resSvcName := checkServiceMatchAndDeleteHeaderKey(req.Service, headers); !match {
```

Comment:
Why do we need to delete the header for this transport?
Let's leave a comment here _or_ above the function definition for `checkServiceMatchAndDeleteHeaderKey(..)`

Reply:
Added, we need delete internal header key before we return response to client.